File: CloudFormation/testSrc/com/intellij/aws/cloudformation/tests/MappingTopLevelKeyReferenceResolveTests.java
package com.intellij.aws.cloudformation.tests;

import com.intellij.aws.cloudformation.references.CloudFormationMappingTopLevelKeyReference;

public class MappingTopLevelKeyReferenceResolveTests extends ResolveTestsBase {
  public MappingTopLevelKeyReferenceResolveTests() {
    super(CloudFormationMappingTopLevelKeyReference.class);
  }

  public void testFindInMapRefToTopLevelKey() throws Exception {
    assertEntityResolve("findInMapRefToTopLevelKey", "m2.xlarge");
  }

  @Override
  protected String getTestDataPath() {
    return TestUtil.getTestDataPath("topLevelKeyResolve");
  }
}

File: CloudFormation/src/test/kotlin/com/intellij/aws/cloudformation/tests/MappingTopLevelKeyReferenceResolveTests.kt
package com.intellij.aws.cloudformation.tests

import com.intellij.aws.cloudformation.references.CloudFormationMappingFirstLevelKeyReference
import java.io.File

class MappingTopLevelKeyReferenceResolveTests : ResolveTestsBase(CloudFormationMappingFirstLevelKeyReference::class.java) {
  fun testFindInMapRefToTopLevelKey() {
    assertEntityResolve("findInMapRefToTopLevelKey.template", "m2.xlarge")
  }

  override val testDataRoot: File
    get() = TestUtil.getTestDataFile("topLevelKeyResolve")
}
